components/script/dom/imagedata.rs

```diff
@@ -111,15 +111,18 @@ impl ImageData {
             reflector_: Reflector::new(),
             width: width,
             height: height,
-            data: Heap::default(),
+            data: HeapTypedArray::default(),
         });
 
         let len = width * height * 4;
         let cx = GlobalScope::get_cx();
+
         rooted!(in (*cx) let mut array = ptr::null_mut::<JSObject>());
-        Uint8ClampedArray::create(*cx, CreateWith::Length(len as usize), array.handle_mut())
-            .unwrap();
-        (*imagedata).data.set(array.get());
+        let typed_array =
```

Comment:
I think this belongs in a new `new_initialized` method of the heaped array, and you should create it above before `imagedata`.

Reply:
And you can make `new_initialized` take an optional `JSObject`, so that it can be used with `new_with_js_object`(see below).